app/controllers/posts_controller.rb

```diff
@@ -28,6 +28,20 @@ def create
     end
   end
 
+  def destroy
+    @user = current_user
+    @post = Post.includes(:likes).find(params[:id])
+    if @post.author == @user || @user.role == 'admin'
+      @post.likes.destroy_all
+      @post.comments.destroy_all
+      @post.destroy!
+      flash[:notice] = 'Post successfully deleted.'
+    else
+      flash[:alert] = 'You are not authorized to delete this Post.'
+    end
+    redirect_to user_posts_path(id: @user.id)
+  end
+
   private
 
   def post_parameters
```

Comment:
- [ ] Good job so far 💪🏿 , please instead of manually loading and authorizing resources in each action, you might want to consider using the `load_and_authorize_resource` method provided by the CanCanCan gem, this  method helps streamline your controller code by automating the resource loading and authorization process. This not only reduces the overall code volume but also makes your controller actions more readable and focused on their specific functionality 👍🏿